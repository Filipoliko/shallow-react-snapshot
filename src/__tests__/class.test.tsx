import React, { Component } from 'react';
import { createPortal } from 'react-dom';
import { shallow } from '../index';
import { render } from '@testing-library/react';

class MyComponent extends Component<{ children?: React.ReactNode, [key: string]: any }> {
    render() {
        const { children, ...props } = this.props;
        return <div id="MyComponent" {...props}>{children || null}</div>;
    }
}

class WrapperAroundMyComponent extends Component<{ children?: React.ReactNode, [key: string]: any }> {
  render() {
      const { children, ...props } = this.props;
      return <MyComponent {...props}>{children || null}</MyComponent>;
  }
}

class MyComponentWithChildrenOnSecondPosition extends Component<{ children?: React.ReactNode }> {
  render() {
      const { children } = this.props;
      return (
        <div id="MyComponentWithChildrenOnSecondPosition">
          <div>First</div>
          {children || null}
        </div>
      );
  }
}

const MyMemoizedComponent = React.memo(MyComponent);

const testCases = [
  {
    description: 'direct wrapper around react component with no child element',
    App: class App extends Component {
      render() {
        return <MyComponent />;
      }
    },
  },
  {
    description: 'direct wrapper around react component with a single child element',
    App: class App extends Component {
      render() {
        return (
          <MyComponent>
            <h1>Hello World One</h1>
          </MyComponent>
        );
      }
    },
  },
  {
    description: 'direct wrapper around react component with a single child element and children on second position',
    App: class App extends Component {
      render() {
        return (
          <MyComponentWithChildrenOnSecondPosition>
            <h1>Hello World One</h1>
          </MyComponentWithChildrenOnSecondPosition>
        );
      }
    },
  },
  {
    description: 'direct wrapper around react component with a single child element and props',
    App: class App extends Component {
      render() {
        return (
          <MyComponent something={'beautiful'}>
            <h1>Hello World One</h1>
          </MyComponent>
        );
      }
    },
  },
  {
    description: 'direct wrapper around react component with multiple deep child elements',
    App: class App extends Component {
      render() {
        return (
          <MyComponent>
            <MyComponent>
              <h1>Hello World One</h1>
            </MyComponent>
          </MyComponent>
        );
      }
    },
  },
  {
    description: 'direct wrapper around react component that wraps another react component',
    App: class App extends Component {
      render() {
        return (
          <WrapperAroundMyComponent>
            <h1>Hello World One</h1>
          </WrapperAroundMyComponent>
        );
      }
    },
  },
  {
    description: 'wrapper around react component with multiple children',
    App: class App extends Component {
      render() {
        return (
          <div>
            <MyComponent>
              <h2>Hello World Two</h2>
              Some random text
            </MyComponent>
          </div>
        );
      }
    },
  },
  {
    description: 'wrapper around react component with single text child',
    App: class App extends Component {
      render() {
        return (
          <div>
            <MyComponent>
              Some random text
            </MyComponent>
          </div>
        );
      }
    },
  },
  {
    description: 'wrapper around react component with single number child',
    App: class App extends Component {
      render() {
        return (
          <div>
            <MyComponent>
              {1234}
            </MyComponent>
          </div>
        );
      }
    },
  },
  {
    description: 'fragment wrapper around react component with single text child',
    App: class App extends Component {
      render() {
        return (
          <>
            <MyComponent>
              <>Some random text</>
            </MyComponent>
          </>
        );
      }
    },
  },
  {
    description: 'direct memoized wrapper around react component with a single child element',
    App: class App extends Component {
      render() {
        return (
          <MyMemoizedComponent>
            <h1>Hello World One</h1>
          </MyMemoizedComponent>
        );
      }
    },
  },
  {
    description: 'wrapped in React.memo',
    App: React.memo(class App extends Component {
      render() {
        return (
          <MyComponent>
            Hello World One
          </MyComponent>
        );
      }
    }),
  },
  {
    description: 'direct profiler wrapper around react component with a single child element',
    App: class App extends Component {
      render() {
        return (
          <React.Profiler id="profiler" onRender={() => {}}>
            <MyComponent>
              <h1>Hello World One</h1>
            </MyComponent>
          </React.Profiler>
        );
      }
    },
  },
  {
    description: 'direct StrictMode wrapper around react component with a single child element',
    App: class App extends Component {
      render() {
        return (
          <React.StrictMode>
            <MyComponent>
              <h1>Hello World One</h1>
            </MyComponent>
          </React.StrictMode>
        );
      }
    },
  },
// @TODO: Suspense is rendering a bit differently for React 16
//   {
//     description: 'direct suspense wrapper around react component with a single child element',
//     App: class App extends Component {
//       render() {
//         return (
//           <React.Suspense fallback={<div>Loading...</div>}>
//             <MyComponent>
//               <h1>Hello World One</h1>
//             </MyComponent>
//           </React.Suspense>
//         );
//       }
//     },
//   },
  {
    description: 'react component with portal',
    App: class App extends Component {
      render() {
        return (
          <MyComponent>
            {createPortal(<h1>Hello World One</h1>, document.body)}
          </MyComponent>
        );
      }
    },
  },
];
  
describe('Class component render', () => {
    test.each(testCases)('$description', ({ App }) => {
        const { container } = render(<App />);

        const result = shallow(container.firstChild as HTMLElement, { whitelist: [App] });

        expect(result).toMatchSnapshot();
    });
});
