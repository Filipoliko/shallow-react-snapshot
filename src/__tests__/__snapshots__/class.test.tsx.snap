// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Class component render direct StrictMode wrapper around react component with a single child element 1`] = `
<StrictMode>
  <MyComponent>
    <h1>
      Hello World One
    </h1>
  </MyComponent>
</StrictMode>
`;

exports[`Class component render direct memoized wrapper around react component with a single child element 1`] = `
<Memo(MyComponent)>
  <h1>
    Hello World One
  </h1>
</Memo(MyComponent)>
`;

exports[`Class component render direct profiler wrapper around react component with a single child element 1`] = `
<Profiler
  id="profiler"
  onRender={[Function]}
>
  <MyComponent>
    <h1>
      Hello World One
    </h1>
  </MyComponent>
</Profiler>
`;

exports[`Class component render direct wrapper around react component that wraps another react component 1`] = `
<WrapperAroundMyComponent>
  <h1>
    Hello World One
  </h1>
</WrapperAroundMyComponent>
`;

exports[`Class component render direct wrapper around react component with a single child element 1`] = `
<MyComponent>
  <h1>
    Hello World One
  </h1>
</MyComponent>
`;

exports[`Class component render direct wrapper around react component with a single child element and children on second position 1`] = `
<MyComponentWithChildrenOnSecondPosition>
  <h1>
    Hello World One
  </h1>
</MyComponentWithChildrenOnSecondPosition>
`;

exports[`Class component render direct wrapper around react component with a single child element and props 1`] = `
<MyComponent
  something="beautiful"
>
  <h1>
    Hello World One
  </h1>
</MyComponent>
`;

exports[`Class component render direct wrapper around react component with multiple deep child elements 1`] = `
<MyComponent>
  <MyComponent>
    <h1>
      Hello World One
    </h1>
  </MyComponent>
</MyComponent>
`;

exports[`Class component render direct wrapper around react component with no child element 1`] = `<MyComponent />`;

exports[`Class component render fragment wrapper around react component with single text child 1`] = `
<MyComponent>
  <Fragment>
    Some random text
  </Fragment>
</MyComponent>
`;

exports[`Class component render react component with portal 1`] = `
<MyComponent>
  <h1>
    Hello World One
  </h1>
</MyComponent>
`;

exports[`Class component render wrapped in React.memo 1`] = `
<MyComponent>
  Hello World One
</MyComponent>
`;

exports[`Class component render wrapper around react component with multiple children 1`] = `
<div>
  <MyComponent>
    <h2>
      Hello World Two
    </h2>
    Some random text
  </MyComponent>
</div>
`;

exports[`Class component render wrapper around react component with single number child 1`] = `
<div>
  <MyComponent>
    1234
  </MyComponent>
</div>
`;

exports[`Class component render wrapper around react component with single text child 1`] = `
<div>
  <MyComponent>
    Some random text
  </MyComponent>
</div>
`;
